File: python/stack_sorter.py
class StackSorter:
    
    def sort(self, stack: list) -> None:
        aux = []

        while len(stack) > 0:
            actual = stack.pop()

            while len(aux) > 0 and aux[-1] > actual:
                stack.append(aux.pop())

            aux.append(actual)

        while len(aux) > 0:
            stack.append(aux.pop())

File: src/utils/StackSorter.java
package utils;

import java.util.Stack;

public class StackSorter {

    public void sort(Stack<Integer> stack) {

        Stack<Integer> aux = new Stack<>();

        while (!stack.isEmpty()) {
            int actual = stack.pop();

            while (!aux.isEmpty() && aux.peek() > actual) {
                stack.push(aux.pop());
            }

            aux.push(actual);
        }

        while (!aux.isEmpty()) {
            stack.push(aux.pop());
        }
    }
}